### DISCUSSION TITLE
Bug Alert! OPP/CobraPin Autofire Coil Firmware

### DISCUSSION THREAD

**Discussion by cobra18t** [missionpinball/missionpinball#48] [General]
We found a bug!

A bug was introduced in OPP firmware version 2.3.0.0 (November 2021) and persisted through 2.3.0.5. Under certain conditions, when you reconfigure an autofire coil, the coil can get stuck on.

This was not the cause of recent coil burning incidents reported on MPF Users, those were wiring related. However, this bug does have the potential to burn a coil, fuse, transistor, etc.

Please visit [OPP Autofire Workaround - Pinball Makers](https://pinballmakers.com/wiki/index.php?title=OPP_Autofire_Workaround) for more details and your workaround options.

**Comment by Pucksnok** [comment]
I'm soon about to make a order with cobra boards again, when do you think new firmware will be done? If I order a satellite board and like three or for new STM32's for my current boards, will they come with the new firmware? Don't have a flasher so prefer getting replacements ordered and shipped when I do a new order.

Cheers.

**Reply by cobra18t** [reply]
The goal is to have the firmware done by the new year, but no guarantees. My pinside shop is on vacation mode until after the new year anyway, so nothing is available for purchase. I intend to reflash all the boards I have in stock with the new firmware and that will be the standard moving forward. Yes, ordering new processor boards is also a way to upgrade to the new firmware.

**Reply by Pucksnok** [reply]
Thx for quick reply, I will wait, I have no rush, looking forward to the updated firmware and a new cobrapin order.

Cheers.

**Comment by cobra18t** [comment]
This bug was fixed in OPP version 2.4.0.0. Thank you Hugh at OPP for getting to this so quickly even during the holidays. I updated [OPP Autofire Workaround - Pinball Makers](https://pinballmakers.com/wiki/index.php?title=OPP_Autofire_Workaround) to include the recommendation to flash 2.4.0.0 as the fix. I re-flashed all my processors so any new orders will get 2.4.0.0.
